ES6/index.html

```diff
@@ -1,115 +1,47 @@
-<!DOCTYPE html>
+<!doctype html>
 <html>
-<head>
+  <head>
     <title>ES6 Demonstration</title>
+    <script>
+      class Paged extends HTMLElement {
+        constructor() {
+          super();
+          const shadow = this.attachShadow({ mode: "open" });
+          let pager = "Page: ";
+          for (let i = 1; i <= 5; i++) {
```

Comment:
The total number of pages should be a value that can be interrogated.

Reply:
By whom? For what purpose? This entire thing is self-contained. yagni.

I don't need to shove this in a state manager or a reducer for no reason, we can put this as a field on the class if you want, but theres no value

Reply:
'reusable' is a significant stretch there, especially when the jQuery example isn't.. Unless by interrogated you mean "implicitly configurable on startup and can't be interrogated" and by reusable you mean that you can run the code multiple times with different arguments. 

Thats now how yagni is applied and you know that... I don't mind storing these in data attributes, but you're making up new requirements on the fly. I think you should layout exactly what the requirements are, it would be more productive than raising rejections

Reply:
Serious question: Why do you say the jQuery code is not reusable?